pkg/kvstore/allocator/allocator.go

```diff
@@ -590,18 +590,35 @@ func (a *Allocator) Get(ctx context.Context, key AllocatorKey) (idpool.ID, error
 	return a.GetNoCache(ctx, key)
 }
 
+func prefixMatchesKey(prefix, key string) bool {
+	// cilium/state/identities/v1/value/label;foo;bar;/172.0.124.60
+	//
+	// labels.SortedList() ensures that the final character in the list of
+	// labels is a semi-colon, which we explicitly leave out of the call
+	// below to strings.TrimRight().
+	k := strings.TrimRight(key, "/0123456789abcdef:")
```

Comment:
I'm not sure if that cutset will work. An alternative would be something like this:
```
k := strings.TrimPrefix(key, prefix)
return k[0] == '/' && strings.Count(k, "/") == 1
```

It would mean that we don't support `/` as part of the node name but it will be liberal otherwise.

Reply:
Ah, is the node name not guaranteed to be an IP, so we try to leave the option open to allow FQDNs in these keys in future or something? Or do you have another case in mind?

Another alternative would be to just do a `strings.Split(key, '/')` and join all but the last. Might be a bit more expensive though.

Reply:
How about this?

```
        lastSlash := strings.LastIndex(key, "/")                                
        return len(prefix) == len(key[0:lastSlash]) 
```